CRM/Contact/BAO/Contact.php

```diff
@@ -2699,9 +2699,8 @@ public static function getValues(&$params, &$values) {
    * @return int|false
    *   total number in database
    */
-  public static function getCountComponent($component, $contactId, $tableName = NULL) {
-    $object = NULL;
-    switch ($component) {
+  public static function getCountComponent(string $type, int $contactId, string $tableName = NULL) {
```

Comment:
the last type hint should be ?string (unless tablename is really compulsory)

Reply:
It already has `default = NULL` so PHP knows it's optional.

Reply:
@colemanw all versions? Because I think type hinting generally expects the ? from what I've seen

Reply:
Yes all versions. When you set a default value of null for the param, the `?` before the type hint is redundant. PHP doesn't care if you include it or not. I've had a hard time finding some authoritative source to say which notation is correcter, should it be `function foo(?string $var = NULL)` or should it be `function foo(string $var = NULL)`. PHP doesn't care and [everyone seems to have their own opinion](https://externals.io/message/98342).

Reply:
@colemanw that link is pretty old - I don't think it stands for php 8 ....

h https://stackoverflow.com/questions/64475517/cannot-use-null-as-default-value-for-parameter-in-php-8

As an aside the tableName property is only used once from core & it would work just fine without it & I found an extension (lineitem editor that passes it in) so it could definitely be deprecated

Reply:
This is the one place https://github.com/civicrm/civicrm-core/pull/22625

Reply:
Ok

Reply:
@eileenmcnaughton I've removed the variable, and noted the nullable info for the future.

Reply:
@colemanw I think we need to deprecate passing in that variable - not remove it - the function is not supported to be called from outside of core. However, lineitemeditor (@monishdeb) and something called 'activityProfile" are both leveraging that `$tableName` so we probably should keep it working, with a deprecation warning

Reply:
Ok done